Add admin pages to the mobile version

SansPseudoFix opened the pull request:
I was starting to code the report view, but I figured out I have to code all admin pages, so…
This is a early-PR because I don't want to code every pages and finally understand it won't be merge or wanted.

So far, reports page, user-search and user entry pages are mobile friendly (doesn't mean totally finished though). I also add admin menu to the drawer.

Here some screenshots:

![drawer](https://cloud.githubusercontent.com/assets/6507951/22311631/9641751a-e354-11e6-871b-f29ebcbab078.png)
![report](https://cloud.githubusercontent.com/assets/6507951/22311633/964cc6d6-e354-11e6-83fb-1777457aaf7e.png)
![usersearch](https://cloud.githubusercontent.com/assets/6507951/22311632/964b1bf6-e354-11e6-9059-2ba237a5ac5d.png)

What do you think? Do I continue or not?

Next:

- [x] dashboard
- [x] weekly user stats
- [x] pod stats
- [x] pod network?
- [x] ~~sidekiq monitor?~~ already mobile friendly
- [x] refactoring/pronto

Edit: and, of course, I forgot my "damit, I forgot pronto" commit… Sorry.

SansPseudoFix committed:
Add admin views to mobile, use partials for common code with desktop

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+
+%li.user.media
+  %div.pull-left
```
`%div.pull-left` can be written as `.pull-left` since `%div` is implicit

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+      %span.media-object
+        = person_image_tag(user.person, size: :thumb_small)
+
+  %div.media-body.row
```
`%div.media-body.row` can be written as `.media-body.row` since `%div` is implicit

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+        = person_image_tag(user.person, size: :thumb_small)
+
+  %div.media-body.row
+    %div.pull-right
```
`%div.pull-right` can be written as `.pull-right` since `%div` is implicit

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+  %div.media-body.row
+    %div.pull-right
+      %span.label.label-default
+        = t('.id')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+        = t('.id')
+        = user.id
+      %span.label.label-info
+        = t('.guid')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+
+    %div
+      .unstyled.text-right.actions
+        = link_to t('admins.user_search.view_profile'), person_path(user.person), class: 'btn btn-default btn-block'
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+    %div
+      .unstyled.text-right.actions
+        = link_to t('admins.user_search.view_profile'), person_path(user.person), class: 'btn btn-default btn-block'
+        = link_to t('admins.user_search.add_invites'), add_invites_path(user.invitation_code), class: 'btn btn-info btn-block'
```
- Line is too long. [126/120]
- Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+        = link_to t('admins.user_search.view_profile'), person_path(user.person), class: 'btn btn-default btn-block'
+        = link_to t('admins.user_search.add_invites'), add_invites_path(user.invitation_code), class: 'btn btn-info btn-block'
+        - unless user.person.closed_account
+          = link_to t('admins.user_search.close_account'), admin_close_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure') }, class: 'btn btn-danger btn-block'
```
- Line is too long. [196/120]
- Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
- Space inside { detected.
- Space inside } detected.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          = link_to t('admins.user_search.close_account'), admin_close_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure') }, class: 'btn btn-danger btn-block'
+
+        - unless user.closed_account?
+          - unless user.access_locked?
```
Do not use `unless` with `else`. Rewrite these with the positive case first.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+
+        - unless user.closed_account?
+          - unless user.access_locked?
+            = link_to t('admins.user_search.lock_account'), admin_lock_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure_lock_account') }, class: 'btn btn-danger btn-block'
```
- Line is too long. [209/120]
- Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
- Space inside { detected.
- Space inside } detected.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          - unless user.access_locked?
+            = link_to t('admins.user_search.lock_account'), admin_lock_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure_lock_account') }, class: 'btn btn-danger btn-block'
+          - else
+            = link_to t('admins.user_search.unlock_account'), admin_unlock_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure_unlock_account') }, class: 'btn btn-danger btn-block'
```
- Line is too long. [215/120]
- Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
- Space inside { detected.
- Space inside } detected.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          - else
+            = link_to t('admins.user_search.unlock_account'), admin_unlock_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure_unlock_account') }, class: 'btn btn-danger btn-block'
+
+    %div.row
```
`%div.row` can be written as `.row` since `%div` is implicit

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+            = link_to t('admins.user_search.unlock_account'), admin_unlock_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure_unlock_account') }, class: 'btn btn-danger btn-block'
+
+    %div.row
+      %div.col-md-5
```
`%div.col-md-5` can be written as `.col-md-5` since `%div` is implicit

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+    %div.row
+      %div.col-md-5
+        %dl.dl-horizontal
+          %dt= t('username')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+        %dl.dl-horizontal
+          %dt= t('username')
+          %dd= user.username
+          %dt= t('.email')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          %dd= user.username
+          %dt= t('.email')
+          %dd= user.email
+          %dt= t('.diaspora_handle')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          %dd= user.email
+          %dt= t('.diaspora_handle')
+          %dd= user.person.diaspora_handle
+          %dt= t('.last_seen')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          %dt= t('.diaspora_handle')
+          %dd= user.person.diaspora_handle
+          %dt= t('.last_seen')
+          %dd= user.last_seen || t('.unknown')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          %dd= user.person.diaspora_handle
+          %dt= t('.last_seen')
+          %dd= user.last_seen || t('.unknown')
+          -if user.invited_by.present?
```
The - symbol should have one space separating it from code

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          %dt= t('.last_seen')
+          %dd= user.last_seen || t('.unknown')
+          -if user.invited_by.present?
+            %dt= t('.invite_token')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          -if user.invited_by.present?
+            %dt= t('.invite_token')
+            %dd= invite_code_url(user.invited_by.invitation_code)
+          %dt= t('.account_closed')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          %dt= t('.account_closed')
+          %dd
+            - if user.person.closed_account
+              %span.label.label-warning= t('.yes')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+            - if user.person.closed_account
+              %span.label.label-warning= t('.yes')
+            - else
+              %span.label.label-success= t('.no')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+              %span.label.label-warning= t('.yes')
+            - else
+              %span.label.label-success= t('.no')
+          %dt= t('.nsfw')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          %dt= t('.nsfw')
+          %dd
+            - if user.person.profile.nsfw
+              %span.label.label-warning= t('.yes')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+            - if user.person.profile.nsfw
+              %span.label.label-warning= t('.yes')
+            - else
+              %span.label.label-success= t('.no')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+            - else
+              %span.label.label-success= t('.no')
+
+        %h4= t('layouts.header.profile')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+        %h4= t('layouts.header.profile')
+
+        %dl.dl-horizontal
+          %dt= t('people.profile_sidebar.born')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+        %dl.dl-horizontal
+          %dt= t('people.profile_sidebar.born')
+          %dd= user.person.profile.birthday
+          %dt= t('people.profile_sidebar.gender')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          %dd= user.person.profile.birthday
+          %dt= t('people.profile_sidebar.gender')
+          %dd= user.person.profile.gender
+          %dt= t('people.profile_sidebar.location')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,70 @@
+          %dd= user.person.profile.gender
+          %dt= t('people.profile_sidebar.location')
+          %dd= user.person.profile.location
+          %dt= t('people.profile_sidebar.bio')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/user_search.mobile.haml:
```diff
@@ -0,0 +1,56 @@
+.col-md-9
+  .row
+    .user_search.col-md-8
+      %h3= t('admins.admin_bar.user_search')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/user_search.mobile.haml:
```diff
@@ -0,0 +1,56 @@
+  .row
+    .user_search.col-md-8
+      %h3= t('admins.admin_bar.user_search')
+      = form_for @search, url: {action: 'user_search'}, html: {method: :get, class: 'form-horizontal'} do |f|
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/user_search.mobile.haml:
```diff
@@ -0,0 +1,56 @@
+      %h3= t('admins.admin_bar.user_search')
+      = form_for @search, url: {action: 'user_search'}, html: {method: :get, class: 'form-horizontal'} do |f|
+        .form-group
+          = f.label :username, t('username'), class: 'col-sm-2 control-label'
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/user_search.mobile.haml:
```diff
@@ -0,0 +1,56 @@
+            = f.text_field :username, class: "form-control"
+
+        .form-group
+          = f.label :email, t('email'), class: 'col-sm-2 control-label'
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/user_search.mobile.haml:
```diff
@@ -0,0 +1,56 @@
+            = f.text_field :email, class: "form-control"
+
+        .form-group
+          = f.label :guid, t('admins.user_entry.guid'), class: 'col-sm-2 control-label'
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

denschub commented:
Woah, please use pronto on your machine, dude.

SansPseudoFix commented:
Yes, I edited my first post: 

> Edit: and, of course, I forgot my "damit, I forgot pronto" commit… Sorry.

Really sorry for that.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,75 @@
+            class: "btn btn-danger btn-block"
+
+        - unless user.closed_account?
+          - unless user.access_locked?
```
Do not use `unless` with `else`. Rewrite these with the positive case first.

SansPseudoFix replied:
I don't know how to fix this

mikestephens replied:
randomly stumbled across this but `- if !user.access_locked?` would be the positive case.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,75 @@
+              data: {confirm: t("admins.user_search.are_you_sure_lock_account")},
+              class: "btn btn-danger btn-block"
+          - else
+            = link_to t('admins.user_search.unlock_account'), admin_unlock_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure_unlock_account') }, class: 'btn btn-danger btn-block'
```
- Line is too long. [215/120]
- Space inside { detected.
- Space inside } detected.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,75 @@
+    .row
+      .col-md-5
+        %dl.dl-horizontal
+          %dt= t('username')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/admins/_user_entry.mobile.haml:
```diff
@@ -0,0 +1,75 @@
+        %dl.dl-horizontal
+          %dt= t('username')
+          %dd= user.username
+          %dt= t('.email')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

Flaburgan commented:
Awesome contribution, thank you! This closes https://github.com/diaspora/diaspora/issues/5076

diaspora-code-review commented on app/views/admins/stats.mobile.haml:
```diff
@@ -0,0 +1,56 @@
+    != t("admins.stats.display_results", segment: content_tag(:strong, @segment))
+
+  .row
+    - [:posts, :comments, :aspect_memberships, :users].each do |name|
```
Use `%i` or `%I` for an array of symbols.

SansPseudoFix replied:
Kind of out of my skills... If someone can help me here, it would be cool.

mikestephens replied:
`%i(posts, comments, aspect_memberships, users)` is what this is looking for.

SansPseudoFix replied:
Seems ok, ty :+1:

diaspora-code-review commented on app/views/admins/stats.mobile.haml:
```diff
@@ -0,0 +1,56 @@
+
+  .row
+    - [:posts, :comments, :aspect_memberships, :users].each do |name|
+      - model = eval("@#{name.to_s}")
```
- The use of `eval` is a serious security risk.
- Redundant use of `Object#to_s` in interpolation.

SansPseudoFix replied:
I don't know what to do here :|

mikestephens replied:
Me randomly again, `eval` in this instance most likely isn't an actual security risk since you are limiting the use of it to your predefined array. Others may have a differing opinion to it though. There isn't really a clean fix for this besides manually writing out each case of posts, comments, etc. Which may be preferred as most ruby devs shy away from eval wherever possible.

cmrd-senya replied:
Try `instance_variable_get("@#{name}")`

SansPseudoFix replied:
For the record, I am not a ruby dev, I only copied/pasted this portion from another file. This is why I can't fix by myself :)
Thx for your help BTW :)

SansPseudoFix replied:
@cmrd-senya Thx! Pronto is happy, as am I :)

diaspora-code-review commented on app/views/admins/weekly_user_stats.mobile.haml:
```diff
@@ -0,0 +1,14 @@
+.col-md-9
+  %h2
+    = t(".current_server", date: Time.now.to_date)
```
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.

SansPseudoFix replied:
If [I understand correctly](https://github.com/diaspora/diaspora_federation/blob/develop/Changelog.md#refactor-1) ``Time.now.iso8601`` is the correct answer?!

diaspora-code-review commented on app/views/admins/weekly_user_stats.mobile.haml:
```diff
@@ -0,0 +1,14 @@
+
+  %div
+    = form_tag("/admins/weekly_user_stats", method: "get", class: "form-inline") do
+      = select_tag(:week, options_for_select(@created_users_by_week.keys.reverse, @selected_week), class: "form-control")
```
Line is too long. [121/120]

diaspora-code-review commented on app/assets/stylesheets/mobile/admins.scss:
```diff
@@ -0,0 +1,14 @@
+#app {
```
Avoid using id selectors

diaspora-code-review commented on app/assets/stylesheets/mobile/admins.scss:
```diff
@@ -0,0 +1,14 @@
+  }
+}
+
+#admin-nav li a {
```
Avoid using id selectors

cmrd-senya commented:
Try something like
instance_variable_get("@#{name}")

SansPseudoFix commented:
Okay, I'm pretty close to the end, but I'm not sure about creating a mobile page for Pod Network. It will be probably very small, no? And, is it useful? Any opinion?

SansPseudoFix commented:
Ready for reviews.

Since I think it's not really mobile friendly to display Pod Network, ``/admins/pods`` displays a warning alert with a link to mobile view.

![mobile_pods_view](https://cloud.githubusercontent.com/assets/6507951/22425720/c8036300-e6fc-11e6-8d0c-45ae79d5a270.png)

Flaburgan commented:
Can you please add the link only on "desktop view", so it is put forward?

SansPseudoFix commented:
Done.
![mobile_pods_view_2](https://cloud.githubusercontent.com/assets/6507951/22426611/37fe497e-e700-11e6-8f42-7492de638b7b.png)

SansPseudoFix committed:
Refactor, more partials + No version check for mobile

SansPseudoFix commented:
New (nested) menu:

![nested_menu](https://cloud.githubusercontent.com/assets/6507951/22468943/9a11314c-e7ca-11e6-9655-c8e9ffc15c89.png)

I didn't notice but Pod status only displays " Determining latest diaspora* version... ", and I think it's a little bit too high level for me to fix it...

SuperTux88 commented:
Sorry that I didn't give feedback earlier. But in my opinion we don't need any of this pages in the mobile interface and we should concentrate on making the "desktop"-interface more responsive. So I think a simple link to the desktop view (like you did on the pods-page) would be enough for all of this pages, so we don't throw a 500er when somebody clicks on the link in the report-email on a smartphone. (Again, just my personal opinion)

But since you already did so much work (thanks for that), why not keeping it. :) But please extract all code that you copied (and maybe improved because of pronto) from the desktop page to partials, that are used by the desktop and mobile interface, so we don't have all of this pages twice with the same a little different code (because of pronto) and need to change two files when we ever change something in the admin-interface.

svbergerem commented on app/views/admins/pods.mobile.haml:
```diff
@@ -0,0 +1,9 @@
+
+      .alert.alert-warning
+        = t(".pod_desktop_view")
+        = link_to t(".pod_desktop_link"), toggle_mobile_path
```
```
= t(".pod_desktop_view", desktop_link: link_to(t(".pod_desktop_link", toggle_mobile_path))
```
or
```
!= t(".pod_desktop_view", desktop_link: link_to(t(".pod_desktop_link", toggle_mobile_path))
```
(I didn't try this) and change the translation to

```
pod_desktop_view: "This page is not available on mobile view, please switch to %{desktop_link}."
```

Please don't concatenate translated strings like this. It gets messy in some languages.

SansPseudoFix replied:
Sorry about that. Fixed.

SansPseudoFix commented:
> But please extract all code that you copied (and maybe improved because of pronto) from the desktop page to partials, that are used by the desktop and mobile interface

I changed many ``div`` and class, so not so many partials (except ``dashboard`` and ``report``), but done.

SuperTux88 commented on app/views/admins/user_search.mobile.haml:
```diff
@@ -0,0 +1,56 @@
+.col-md-9
```
This page is also nearly the same, the only difference is `pull-right` vs `btn-block` for the submit buttons. This can be done for example with a `button_class` parameter for the partial.

SuperTux88 reviewed (changes requested):
I found two more partials that can be created with little effort. Also we want to make the desktop-interface more responsive, so it works on mobile devices too, so maybe you can do this for the desktop-interface, and then use the desktop code also on mobile, without changing it.

Sorry, but I just don't want so much code duplication and then change 3 lines, only to add something to the mobile interface, when we try to make the desktop-interface work on mobile too, instead of adding more stuff to the mobile-interface.

I'm fine with adding this to the mobile-interface, when you can do it without duplicating everything.

SansPseudoFix commented:
Is it better?

Flaburgan commented:
https://github.com/diaspora/diaspora/blob/develop/app/assets/javascripts/app/pages/admin_dashboard.js is responsible for the version checking though I don't know how you can include it in the mobile version because it doesn't use backbone as far as I know... @svbergerem what's the cleanest solution here?

svbergerem commented:
> @svbergerem what's the cleanest solution here?

For this PR I'd say that we could start with a dashboard without the version check.

SansPseudoFix commented:
I removed version check for mobile and rebased.

Flaburgan reviewed (changes requested):
I only did a "user point of view" review at the moment:
- The admin home page "dashboard" only displays "Pod status" without anything now that you removed the version check. It feels weird. I will see if I can include the version check (or help you to do so)
- The user search page displays the search results under the "invite" section. On a phone, the user doesn't see the results without scrolling so he has the feeling the search didn't work. It would be better to display the invite section **after** the results
- When the admin invites a user, the confirmation message isn't displayed. @svbergerem touched the messages on the mobile version recently, maybe he can advice about that?

Everything else seems to work correctly. Nice job!

SansPseudoFix commented:
> The admin home page "dashboard" only displays "Pod status" without anything now that you removed the version check. It feels weird. I will see if I can include the version check (or help you to do so)

Out of my skills. But it can be done later.

> The user search page displays the search results under the "invite" section. On a phone, the user doesn't see the results without scrolling so he has the feeling the search didn't work. It would be better to display the invite section after the results

I tried with bootstrap classes, but nothing works. I don't think this is a big issue, but if you have a solution…

Flaburgan committed:
Move invitation form under the user search in admin

Indicates when there is no report to review to avoid a weird empty page

Flaburgan committed:
Add mobile message confirmation on user invit action, Refactor admin responsive code to use bootstrap grid

Hide pod version alert only in the mobile version, not depending on screen size

Flaburgan commented:
I fixed my 3 remarks with 3 commits. The code is rebased and I reviewed it, this looks fine to me now :)
@SuperTux88 if you have opinion about the partials (as the requested changes have been done) otherwise this is ready to be reviewed by someone else.

diaspora-code-review commented on app/views/report/_reports.haml:
```diff
@@ -0,0 +1,42 @@
+  #reports
+    %h1
+      = t("report.title")
+    - if @reports.size == 0
```
Use `empty?` instead of `size == 0`.

diaspora-code-review commented on app/views/report/_reports.haml:
```diff
@@ -0,0 +1,42 @@
+          - username = report.user.username
+          .panel-heading
+            .reporter.pull-right
+              = raw t("report.reported_label", person: link_to(username, user_profile_path(username)))
```
Tagging a string as html safe may be a security risk, prefer `safe_join` or other Rails tag helpers instead.

diaspora-code-review commented on app/views/report/_reports.haml:
```diff
@@ -0,0 +1,42 @@
+          - username = report.user.username
+          .panel-heading
+            .reporter.pull-right
+              = raw t("report.reported_label", person: link_to(username, user_profile_path(username)))
```
Tagging a string as html safe may be a security risk, prefer `safe_join` or other Rails tag helpers instead.

diaspora-code-review commented on app/assets/stylesheets/mobile/admins.scss:
```diff
@@ -0,0 +1,28 @@
+}
+
+@media(max-width: $screen-xs-max) {
+  #app {
```
Avoid using id selectors

diaspora-code-review commented on app/views/admins/_user_entry.haml:
```diff
@@ -1,70 +1,87 @@
+      .row
+        .col-md-7
+          %h4.media-heading
+            = user.person.name if user.person
```
Use safe navigation (`&.`) instead of checking if an object exists before calling the method.

Flaburgan replied:
Is `user&.person&.name` what's waited here?

diaspora-code-review commented on app/views/admins/_user_entry.haml:
```diff
@@ -1,70 +1,87 @@
+              = user.id
+            %span.label.label-info
+              = t(".guid")
+              = user.person.guid if user.person
```
Use safe navigation (`&.`) instead of checking if an object exists before calling the method.

diaspora-code-review commented on app/views/admins/_user_entry.haml:
```diff
@@ -1,70 +1,87 @@
+            %dd= user.person.diaspora_handle
+            %dt= t(".last_seen")
+            %dd= user.last_seen || t(".unknown")
+            -if user.invited_by.present?
```
The - symbol should have one space separating it from code

diaspora-code-review commented on app/views/admins/_user_entry.haml:
```diff
@@ -1,70 +1,87 @@
-          %dt= t('people.profile_sidebar.bio')
-          %dd= user.person.profile.bio
+            - unless user.closed_account?
+              - unless user.access_locked?
```
Do not use `unless` with `else`. Rewrite these with the positive case first.

Flaburgan commented:
Everything is squashed, code is fixed. I did some refactor to use bootstrap grid.
It is now ready to be reviewed. (Not sure where to click in github to request a review by @svbergerem )

Flaburgan commented:
`bin/rspec spec/controllers/admins_controller_spec.rb` is green locally for me, does anyone know what could make it red here?

SuperTux88 reviewed (approved):
> @SuperTux88 if you have opinion about the partials (as the requested changes have been done) 

The structure with the partials look good now, no more code duplication 👍 But I didn't do any deeper code reviews or testing, so that's still needed to be done, just approving my requested changes. :)

SuperTux88 commented:
> `bin/rspec spec/controllers/admins_controller_spec.rb` is green locally for me, does anyone know what could make it red here?

That's because the "Safe Navigation Operator" (`&.`) is only available in ruby 2.3 and newer. For older versions you can use `try` in rails: `user&.person&.name` -> `user.try(:person).try(:name)`

SansPseudoFix commented on app/views/admins/_user_entry.haml:
```diff
@@ -1,70 +1,85 @@
+              = user.id
+            %span.label.label-info
+              = t(".guid")
+              = user.try(:person).try(:guid)
```
Is it okay like this?

Flaburgan replied:
You could have try locally to switch Ruby to 2.1 and run rspec but Travis will tell us soon ;)

svbergerem commented on app/views/report/_reports.haml:
```diff
@@ -0,0 +1,42 @@
+              class: "btn pull-left btn-info btn-small col-md-3 col-xs-12", method: :put
+            = button_to t("report.delete_link"), report_path(report.id, type: report.item_type),
+              data: {confirm: t("report.confirm_deletion")},
+              class: "btn pull-right btn-danger btn-small col-md-3 col-xs-12", method: :delete
```
**width: 360px**
![reports360](https://cloud.githubusercontent.com/assets/3798614/24608212/e8f4b366-1875-11e7-8c46-87f584abc5e2.png)

SansPseudoFix commented on app/views/admins/_admin_bar.haml:
```diff
@@ -1,9 +1,7 @@
   = stylesheet_link_tag :admin
 
-%h2= t(".pages")
-
```
Forgotten mistake. Restored.

Flaburgan commented on app/views/admins/_user_entry.haml:
```diff
@@ -1,70 +1,85 @@
+    .col-md-1
+      - if user.person
+        %span.media-object
+          = person_image_tag(user.person, size: :thumb_small)
```
I tried to fix that with https://github.com/Flaburgan/diaspora/commit/b7bde63be75a0d5e653fa857864d35b35ae48615#diff-9d44402b8caede0852040adab1428922R5 but the avatar is not displayed anymore even if it has the 50x50 size fixed by CSS as before. Any idea what's going on?

Flaburgan commented:
Alright remarks should be fixed. There is room for improvements in the markup, but I think more refactor should not be done here.

cmrd-senya commented on app/views/admins/dashboard.mobile.haml:
```diff
@@ -0,0 +1,4 @@
+- content_for :head do
+  = stylesheet_link_tag :admin
```
Do we really need linking this stylesheet here? In the current version it isn't included in desktop version and no elements in dashboard (both versions) really seem to rely on the styles defined in `admin.css`.

cmrd-senya commented on app/views/admins/dashboard.mobile.haml:
```diff
@@ -0,0 +1,4 @@
+- content_for :head do
+  = stylesheet_link_tag :admin
+
+= render partial: "admins/dashboard"
```
In mobile Dashboard now I have a message " Determining latest diaspora* version... " that is never actually get replaced with anything else. In desktop version it determines the newest version and changes its background to green. If the version check doesn't work for mobile these "Determining latest diaspora* version... " words perhaps shouldn't be rendered or it should say "Cannot determine latest diaspora* version" and maybe just display the current version.

Flaburgan replied:
This has the `.hidden-xs` class so it is not displayed on mobile. But you're right this hack is not nice, because large devices on mobile version still have the message, and mobile devices with desktop version don't have it anymore. I'll do it with CSS to hide it if the user is using the mobile version, not depending on his screen size.

cmrd-senya reviewed (changes requested):
I reviewed some parts of the PR. I haven't finished the review since I don't have more time tonight, but I'll get back to this later. Here is what I've discovered.

cmrd-senya commented on app/views/report/_reports.haml:
```diff
@@ -0,0 +1,46 @@
+- content_for :head do
+  = stylesheet_link_tag :admin
+  = stylesheet_link_tag :report
```
You have deleted `report.scss` so there is no need in this line anymore

cmrd-senya commented on app/views/report/_reports.haml:
```diff
@@ -0,0 +1,46 @@
+            .segment-selection
+              = button_to t("report.reported_user_details"),
+                user_search_path(admins_controller_user_search: {guid: report.reported_author.guid}),
+                class: "btn pull-left btn-info btn-small col-md-3 col-xs-12", method: :post
```
The buttons should have a space in between:
![2017-04-28 15-32-56](https://cloud.githubusercontent.com/assets/10187586/25528868/5b3b7688-2c28-11e7-87c6-40d3f95832f3.png)

cmrd-senya reviewed (changes requested):
Here is another portion of my findings

cmrd-senya commented:
Also, I don't know where it belongs to, but mobile admin pages have some free space on the right. Perhaps we should stretch the content to the whole screen width (like we do for non-admin mobile pages like streams). What do you think?

![2017-04-28 15-37-08](https://cloud.githubusercontent.com/assets/10187586/25529302/2a2f2452-2c2a-11e7-88f9-3977bf367ec4.png)

cmrd-senya commented:
Also, we don't have any tests for admin pages and I don't think we really need to test it too deep, but since it is fix for #5076, it would be nice to add at least a simple test to verify it isn't `500` anymore. That probably can be achieved by extending `ReportController` unit test.

cmrd-senya reviewed (changes requested):
Okay, that's the last portion of my notes for this review round.

Flaburgan committed:
Refactor partials + pronto

Flaburgan commented on app/views/layouts/_drawer.mobile.haml:
```diff
@@ -35,5 +35,22 @@
+            %li= link_to t("admins.admin_bar.user_search"), user_search_path
+            %li= link_to t("admins.admin_bar.weekly_user_stats"), weekly_user_stats_path
+            %li= link_to t("admins.admin_bar.pod_stats"), pod_stats_path
+            %li{role: "presentation", class: current_page?(report_index_path)}
```
Indeed, what did you try to achieve here @SansPseudoFix ?

SansPseudoFix replied:
It's a copy/paste from [desktop view's menu](https://github.com/diaspora/diaspora/blob/795283b581d6e9d2ec93f150a00a8efe90779814/app/views/admins/_admin_bar.haml#L6). My ruby/HAML skills are **really** limited, so I try many things, and when it works, I don't touch anymore.

BTW, thank you very much for your review/time @cmrd-senya!

Flaburgan replied:
Alright, removing then.

Flaburgan commented on app/views/admins/_user_entry.haml:
```diff
@@ -1,70 +1,83 @@
+            method: :post, data: {confirm: t("admins.user_search.are_you_sure")},
+            class: "btn btn-danger btn-block"
+
+        - unless user.closed_account?
```
What difference is there between `unless user.person.closed_account` and `unless user.closed_account?`? I did write this code I just refactored the markup so not sure I should change anything here.

cmrd-senya replied:
There is no difference, `user.closed_account?` delegates to  `user.person.closed_account`: https://github.com/diaspora/diaspora/blob/7eb3dabb444d99cf51e672e065284f9aa23628e9/app/models/user.rb#L514-L514

cmrd-senya replied:
Again, you left it as is while you could just removed lines 70 and 71. Why?

cmrd-senya replied:
> I did write this code I just refactored the markup so not sure I should change anything here.

I don't really think it matters who has written the code. Even though you copy pasted it, technically you just removed the code from one place and added it to another. There is no difference between adding the code that you made up in your head and adding the code you copied over from some example. If you prosose some additions they shouldn't contain any lame code, even though you copied it from somewhere. That's why we have code reviews - to avoid introduction (or reintroduction) of lame code. That's my opinion. Of course, I can't force you, so decide it for yourself.

Flaburgan commented:
@cmrd-senya I corrected most of your remarks, some of them were concerning copy pasted code I'd like to avoid to modify. This PR is already huge enough.

Ready for another round of review. It would be very nice to see this included in 0.6.6.0 which lakes new features at the moment.

cmrd-senya commented:
You still didn't add any tests as I proposed at https://github.com/diaspora/diaspora/pull/7295#issuecomment-298084099.

Don't you mind adding at least a simple test to verify that the mobile reports page isn't `500` anymore?

cmrd-senya commented on app/views/admins/_stats.haml:
```diff
@@ -0,0 +1,60 @@
+  = form_tag("/admins/stats", method: "get", class: "form-inline segment-selection") do
+    .col-sm-2.col-sm-push-8
+      %select.form-control{name: "range"}
+        %option{value: "daily", selected: ("selected" if params[:range] == "daily")}
```
I'm quite sad to see you didn't apply the changes I proposed at https://github.com/diaspora/diaspora/pull/7295#discussion_r114014088 and https://github.com/diaspora/diaspora/pull/7295#discussion_r114016339. I know that this is a copy-pasted code, but I see no reasons to copy over lame code and to not introducing these little fixes that make that PR smaller, the code more readable and which are completely safe to make (it's just about code style).

Flaburgan replied:
I'll do it. My point here is I usually don't refactor code I don't understand, I don't want to break stuff or have any side effects. So as a beginner I prefer to be careful.

cmrd-senya commented on app/views/admins/_user_search.haml:
```diff
@@ -0,0 +1,58 @@
+          .col-xs-12.col-md-8
+            = text_field_tag "identifier", nil, class: "form-control"
+          .col-xs-12.col-md-2
+            = submit_tag t(".invite"), class: "btn btn-block btn-primary"
```
This one is not fixed

cmrd-senya commented on app/assets/stylesheets/admin.scss:
```diff
@@ -32,9 +31,34 @@
+.segment-selection {
+  select,
+  [type=submit] {
+    margin-top: 10px;
```
Do we need this css rule actually? When I turn it off it feels better, aligned with "Displaying results from the Daily segment" line. For example, for `/admins/weekly_user_stats` we align the selection box with the heading. So why having it lower here?

Flaburgan replied:
It avoids the button to touch the selector. The focus here is the mobile view ;)

cmrd-senya replied:
How is it actually different from `.week-selector`, where `margin-top` is `18px`?

Flaburgan replied:
It's not. I made `.week-selector` `margin-top: 10px` too. Note that I didn't spend much time building a nice design, there are still inconsistencies, some titles are `h1` when others are `h2` for example. The markup can be refactored to something cleaner but those are admin pages and I didn't want to spend too much time rewriting them. The focus here is to make them available on mobile with a usable interface.

cmrd-senya replied:
But `margin-top` is set to 18px here https://github.com/diaspora/diaspora/pull/7295/files#diff-111bd41e9b25b9eaafc015184274028dR40

Flaburgan replied:
Not anymore :p

cmrd-senya commented on app/controllers/admin/pods_controller.rb:
```diff
@@ -26,6 +27,7 @@ def recheck
       respond_with do |format|
         format.html { redirect_to admin_pods_path }
         format.json { render json: PodPresenter.new(pod).as_json }
+        format.mobile { redirect_to admin_pods_path }
```
The `recheck` request is not available on mobile, so this line is redundant.

diaspora-code-review commented on app/views/admins/_stats.haml:
```diff
@@ -0,0 +1,50 @@
+  - %i(posts comments aspect_memberships users).each do |name|
+    - model = instance_variable_get("@#{name}")
+    - if name == :aspect_memberships
+      - name = t(".shares", count: model[:yesterday])
```
3 consecutive Ruby scripts can be merged into a single `:ruby` filter

diaspora-code-review commented on app/views/admins/_stats.haml:
```diff
@@ -0,0 +1,50 @@
+
+  .col-sm-8.col-sm-pull-4
+    %h4
+      != t("admins.stats.display_results", segment: content_tag(:strong, @segment))
```
Avoid using instance variables in partials views

Flaburgan replied:
Can someone tell me how I should refactor that to solve this warning?

SuperTux88 replied:
Include the partial like this:

```haml
= render partial: "admins/stats", locals: {segment: @segment}
```

and then use `segment` here instead of `@segment`

diaspora-code-review commented on app/views/admins/_stats.haml:
```diff
@@ -0,0 +1,50 @@
+
+.row
+  %p.col-md-12.alert.alert-info.text-center{role: "alert"}
+    != t("admins.stats.current_segment",
```
Avoid using instance variables in partials views

diaspora-code-review commented on app/views/admins/_user_search.haml:
```diff
@@ -0,0 +1,57 @@
+.row
+  .col-md-12
+    .alert.alert-info.text-center{role: "alert"}
+      = t(".users", count: @users.count)
```
Avoid using instance variables in partials views

diaspora-code-review commented on app/views/admins/_weekly_user_stats.haml:
```diff
@@ -0,0 +1,21 @@
+  .col-md-3.col-xs-12
+    = form_tag("/admins/weekly_user_stats", method: "get", class: "form-inline week-selector row") do
+      .col-xs-9.center
+        = select_tag(:week,
```
Avoid using instance variables in partials views

diaspora-code-review commented on app/views/admins/_weekly_user_stats.haml:
```diff
@@ -0,0 +1,21 @@
+      .col-xs-3.center
+        = submit_tag t("admins.stats.go"), class: "btn btn-primary"
+
+= t(".amount_of", count: @counter)
```
Avoid using instance variables in partials views

Flaburgan committed:
Add tests for mobile drawer

Flaburgan commented on app/views/admins/_stats.haml:
```diff
@@ -0,0 +1,50 @@
+  = form_tag("/admins/stats", method: "get", class: "col-sm-12 form-inline segment-selection") do
+    .col-sm-2.col-sm-push-8
+      %select.form-control{name: "range"}
+        - ["daily", "week", "2weeks", "month"].each do |range|
```
Pronto is asking me to write that line with `%w(daily week 2weeks month).each do |range|` but then the code crashes saying "Illegal nesting: content can't be both given on the same line as %w and nested within it." in the line below (`%option`)

SuperTux88 replied:
What do you mean with "the code crashes"? I tested this and it doesn't crash:

```diff
--- a/app/views/admins/_stats.haml
+++ b/app/views/admins/_stats.haml
@@ -7,7 +7,7 @@
   = form_tag("/admins/stats", method: "get", class: "col-sm-12 form-inline segment-selection") do
     .col-sm-2.col-sm-push-8
       %select.form-control{name: "range"}
-        - ["daily", "week", "2weeks", "month"].each do |range|
+        - %w(daily week 2weeks month).each do |range|
           %option{value: range, selected: ("selected" if params[:range] == range)}
             = t(".#{range}")
     .col-sm-2.col-sm-push-8
```

Flaburgan replied:
Sorry, I thought % was the same than for the HTML so I removed the `-`. Now it works.

Flaburgan committed:
Fix design issues + pronto

Flaburgan commented:
This should be ready for another round of review.

cmrd-senya commented:
https://github.com/diaspora/diaspora/pull/7295#discussion_r114025807 this one is still not fixed

cmrd-senya commented on app/views/admins/_weekly_user_stats.haml:
```diff
@@ -0,0 +1,21 @@
+
+= t(".amount_of", count: counter)
+%br
+- @created_users_by_week[selected_week].each do |m|
```
should be `created_users_by_week` instead of `@created_users_by_week`

Flaburgan replied:
Doing that I can an error *no implicit conversion of String into Integer*, not sure how to solve it, I tried to print `selected_week` but it looks it is empty. Weird.

cmrd-senya replied:
Works for me the following way: https://gist.github.com/cmrd-senya/8378adf3ad3e5e059ae1fc1a8128f56f

Flaburgan replied:
Thanks got it. I did it with only one variable. So, corrected.

cmrd-senya reviewed (changes requested):
Here are a few notes after a quick scan. I'll do a full review  round after you respond to them.

Flaburgan commented:
So still a question, all other points should be resolved.

cmrd-senya commented on app/assets/stylesheets/admin.scss:
```diff
@@ -16,14 +16,17 @@
   }
 }
 
+.invite-emails {
```
I think this should go below the `/** Invites panel **/` comment

cmrd-senya reviewed (commented):
Okay, I reviewed the PR once again. Everything generally looks good to me. There are a few more comments, which are mostly nit-picking. IMO it would be nice to fix them. I can approve the PR just after this is sorted out.

SuperTux88 commented on app/views/admins/_user_entry.haml:
```diff
@@ -1,70 +1,81 @@
+          %dt= t(".email")
           %dd= user.email
-          %dt= t('.diaspora_handle')
+          %dt= t(".diaspora_handle")
```
Yes, the key shouldn't be changed (otherwise we need to re-translate it), but the translation should be "diaspora\* ID".

Flaburgan commented:
Remarks fixed, thanks for your review!

cmrd-senya reviewed (approved):
Okay, looks good to me now. I don't know if anyone else will review, but note that before merge some of the commits should be squashed.

SuperTux88 commented:
I just had a look at this (because I wanted to check if I can merge it to 0.6.7.0), and I think the code is OK and it works. There are probably things that can be done better (for example I'd like when the admin section stays open in the drawer when at an admin page, so I don't need to open it every time), but this can always be improved in a follow up. So it's OK for now (this PR is already big enough), and I don't want to polish mobile admin that much anyway, because I think we should focus on making the desktop view more responsive.

But there are really many commits, so as @cmrd-senya wrote, it would be good if you can squash some of them (so fix pronto stuff and review remarks in the commit where it belongs). So can you try to squash some of them, because when trying to find where some lines come from and why they are the way they are and then find something like "Fix remarks" with some random changes in it doesn't help ;) When it's too complicated and you have many conflicts, then just let it as it is, instead of breaking everything when resolving conflicts.

Flaburgan commented:
I was waiting for a final "Everything is fine" to squash ;) I'll do it today.

Flaburgan commented:
Alright, commits squashed from 28 down to 7. Please tell me if that's enough.

Flaburgan commented:
I just realized that this PR doesn't deal with the moderator role. I guess it's not a blocker I'll do it in another one.

SuperTux88 reviewed (approved):
> Alright, commits squashed from 28 down to 7. Please tell me if that's enough.

Yes 👍

SuperTux88 commented:
Merged as 07d0ed916dcaeeebfe38b5315f1d4160a449341c, 427aa8794e36c568caa5efc070864b45ab0d56e6, 2782edc780f9c737a6d5d459c51cc741ae632909, a7d97b7fd3f6f6455f8b04f4eea09b2c2d2ea623, 7bdf33e1a8cf961ad52bd5bed257ab1f35e1ae50, 5cb4c6e5df044eec6c5f49d16b1c3662671520ac and 276b640418f8cefb103d3e8003f4d264793bb35d

Thank you @SansPseudoFix and @Flaburgan and also thank you @cmrd-senya for the reviews ❤️
